src/CMakeLists.txt

```diff
@@ -684,48 +684,48 @@ if (${NUPIC_BUILD_PYEXT_MODULES})
   # capnproto symbols by preloading the pycapnp extension in our extension
   # python proxy modules, thus avoiding the conflict from executing the methods
   # in our own compilation of capnproto on objects created by pycapnp's.
-  set(src_swig_link_libraries
+  set(_SRC_SWIG_LINK_LIBRARIES
       ${src_lib_static_nupiccore_combined}
       ${src_common_os_libs})
 
   # Common dependencies for our python extensions for use with
   # SWIG_MODULE_name_EXTRA_DEPS
-  set(src_swig_extra_deps Swig)
+  # Make sure we don't execute the swig executable before it is built
+  set(_SRC_SWIG_EXTRA_DEPS Swig)
 
   # NOTE Windows DLLs are shared executables with their own main; they require
   # all symbols to resolve at link time, so we have to add libpython for this
   # platform
   #
   # NOTE On Windows nupic.bindings builds, we include capnproto because we
   # presently build self-contained CAPNP_LITE on Windows, and Windows
-  # nupic/nupic.bindings presently have no dependency on pycapnp. We also include
-  # PYTHON_LIBRARIES because all symbols have to resolve when building a DLL.
+  # nupic/nupic.bindings presently have no dependency on pycapnp.
   if("${PLATFORM}" STREQUAL "windows")
-    list(APPEND src_swig_link_libraries
+    list(APPEND _SRC_SWIG_LINK_LIBRARIES
          ${PYTHON_LIBRARIES}
          ${CAPNP_STATIC_LIB_TARGET})
   endif()
 
-  message(STATUS "src_swig_extra_deps     = ${src_swig_extra_deps}")
-  message(STATUS "src_swig_link_libraries = ${src_swig_link_libraries}")
-  message(STATUS "src_swig_extension_link_flags= ${src_swig_extension_link_flags}")
+  message(STATUS "_SRC_SWIG_EXTRA_DEPS     = ${_SRC_SWIG_EXTRA_DEPS}")
+  message(STATUS "_SRC_SWIG_LINK_LIBRARIES = ${_SRC_SWIG_LINK_LIBRARIES}")
+  message(STATUS "_SRC_SWIG_EXTENSION_LINK_FLAGS= ${_SRC_SWIG_EXTENSION_LINK_FLAGS}")
   message(STATUS "CMAKE_SWIG_FLAGS        = ${CMAKE_SWIG_FLAGS}")
 
 
   function(PREPEND_BOILERPLATE_TO_PYTHON_PROXY_MODULE
```

Comment:
@scottpurdy, I added this function in an earlier PR. The other day, I got a response to my question on the Swig mailing list about `%pythonbegin` that looks like could be used to accomplish the same feat. I filed issue #1043  as a reminder to follow up.